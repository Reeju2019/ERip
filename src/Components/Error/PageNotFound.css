.errorImg {
  max-width: 20vw;
}

.notFoundCard {
  width: 50vw;
  margin: auto;
}

.cardInner {
  border: 1px solid rgb(92, 92, 92);
  border-radius: 20px;
  padding: 2vmax;
  box-shadow: 0px 0px 30px rgb(92, 92, 92);
  background-color: white;
}
.cont {
  background-color: rgb(250, 250, 150);
  height: 100vh;
  width: 100vw;
}
.notFoundBtn {
  border: 1px solid black;
  border-radius: 5px;
  padding: 5px;
  width: 12vmax;
  transition: all 0.5s;
}

.notFoundBtn:hover {
  background: linear-gradient(to right, rgb(209, 113, 113), rgb(255, 255, 255));
  color: gray;
  border: 1px solid green;
}
