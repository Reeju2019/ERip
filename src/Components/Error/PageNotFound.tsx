import React from 'react'
import { Container } from 'react-bootstrap'
import './PageNotFound.css'
import { useNavigate } from 'react-router-dom'

const PageNotFound: React.FunctionComponent = () => {
  const navigate = useNavigate()
  const handleClick = () => {
    navigate('/')
  }
  return (
    <>
      <Container fluid className='p-5 cont'>
        <div className='notFoundCard'>
          <div className='cardInner'>
            <div className='d-flex justify-content-between'>
              <div className='fs-1'>
                <div>OOPS!</div>
                <div>Page Not Found</div>
                <div>ðŸ¥º</div>
              </div>
              <div>
                <img src='/static/errorPage/giff.gif' className='errorImg' alt='' />
              </div>
            </div>
          </div>
          <div className='d-flex justify-content-center m-4'>
            <button onClick={handleClick} type='button' className='notFoundBtn'>
              Go to Home?
            </button>
          </div>
        </div>
      </Container>
    </>
  )
}

export default PageNotFound
