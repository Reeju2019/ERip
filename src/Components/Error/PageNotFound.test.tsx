import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import PageNotFound from './PageNotFound'

test('page not found', () => {
  render(<PageNotFound />)
})

describe('pnf rendering specification', () => {
  it('SnapShot testing', () => {
    const component = renderer.create(<PageNotFound />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render without crashing', () => {
    const div = document.createElement('div')
    render(<PageNotFound />)
    ReactDOM.unmountComponentAtNode(div)
  })
})
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}))
