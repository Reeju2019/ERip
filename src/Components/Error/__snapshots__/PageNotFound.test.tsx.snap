// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pnf rendering specification SnapShot testing 1`] = `
<div
  className="p-5 cont container-fluid"
>
  <div
    className="notFoundCard"
  >
    <div
      className="cardInner"
    >
      <div
        className="d-flex justify-content-between"
      >
        <div
          className="fs-1"
        >
          <div>
            OOPS!
          </div>
          <div>
            Page Not Found
          </div>
          <div>
            ðŸ¥º
          </div>
        </div>
        <div>
          <img
            alt=""
            className="errorImg"
            src="/static/errorPage/giff.gif"
          />
        </div>
      </div>
    </div>
    <div
      className="d-flex justify-content-center m-4"
    >
      <button
        className="notFoundBtn"
        onClick={[Function]}
        type="button"
      >
        Go to Home?
      </button>
    </div>
  </div>
</div>
`;
